zlog_test.go

```diff
@@ -12,6 +12,51 @@ import (
 	"go.opentelemetry.io/otel/baggage"
 )
 
+func TestEscape(t *testing.T) {
+	t.Run("NeedEscape", func(t *testing.T) {
+		tt := []struct {
+			In   string
+			Want bool
+		}{
+			{"%25", false},
+			{"%2C", false},
+			{"%age", true},
+			{"%", true},
+			{"ꙮ", true},
+			{"%", true},
```

Comment:
nit: already tested?

Reply:
I wanted to test specifically that the regexp omits already-escaped values, but catches all the ones we need to escape.